dissect/target/plugins/browsers/chromium.py

```diff
@@ -23,76 +25,81 @@ class ChromiumMixin:
+
+        Returns:
+            list with tuples containing user and TargetPath object.
+        """
+        users_dirs: list[tuple] = []
+        for user_details in self.target.user_details.all_with_home():
+            for d in hist_paths:
+                cur_dir: TargetPath = user_details.home_path.joinpath(d)
+                cur_dir = cur_dir.resolve()
+                if not cur_dir.exists() or (user_details.user, cur_dir) in users_dirs:
+                    continue
+                users_dirs.append((user_details.user, cur_dir))
+        return users_dirs
+
+    def _iter_db(self, filename: str) -> Iterator[SQLite3]:
+        """Generate a connection to a sqlite history database file.
+
+        Args:
+            filename: The filename as string of the database where the history is stored.
         Yields:
-            Records with the following fields:
-                hostname (string): The target hostname.
-                domain (string): The target domain.
-                ts (datetime): Visit timestamp.
-                browser (string): The browser from which the records are generated from.
-                id (string): Record ID.
-                url (uri): History URL.
-                title (string): Page title.
-                description (string): Page description.
-                rev_host (string): Reverse hostname.
-                visit_type (varint): Visit type.
-                visit_count (varint): Amount of visits.
-                hidden (string): Hidden value.
-                typed (string): Typed value.
-                session (varint): Session value.
-                from_visit (varint): Record ID of the "from" visit.
-                from_url (uri): URL of the "from" visit.
-                source: (path): The source file of the history record.
+            opened db_file (SQLite3)
         Raises:
-            SQLError: If the history file could not be processed.
+            FileNotFoundError: If the history file could not be found.
+            SQLError: If the history file could not be opened.
         """
-        for user, db_file, db in self._iter_db("History"):
+
+        for user, cur_dir in self._build_userdirs(self.DIRS):
+            db_file = cur_dir.joinpath(filename)
             try:
-                urls = {row.id: row for row in db.table("urls").rows()}
-                visits = {}
+                yield user, db_file, sqlite3.SQLite3(db_file.open())
+            except FileNotFoundError:
+                self.target.log.warning("Could not find %s file: %s", filename, db_file)
+            except SQLError as e:
+                self.target.log.warning("Could not open %s file: %s", filename, db_file, exc_info=e)
 
-                for row in db.table("visits").rows():
-                    visits[row.id] = row
-                    url = urls[row.url]
+    def _iter_json(self, filename: str) -> Iterator[tuple[str, TargetPath, dict]]:
```

Comment:
```suggestion
    def _iter_json(self, filename: str) -> Iterator[tuple[str, Path, dict]]:
```

Reply:
Yes, but everywhere else in `dissect.target` we typehint `Path` instead of `TargetPath`, just to provide a more generic type hint. Similar to how you typehint ` fh` as `BinaryIO` instead of `AlignedStream`. There's not really any specific reason to typehint as specific as `TargetPath` IMO.

Reply:
For input parameters, if you don't use the `TargetPath` specific bits I agree. For output types, I don't. Here it does matter that you don't get a `Path` but a `TargetPath` as it will have different properties and behave slightly different from `Path`. Now for instance I need to trace back the code to make sure I actually have the `._fs` property available when I get back the path.

Granted in this case it is not needed because the consumer of this function does not use the `TargetPath` specifics,. But as a function you'll never know who your consumers will be.

Reply:
Fair argument, and I agree. Let's change the return value to `TargetPath` in that case and try to maintain that in other places too.